src/OrchardCore/OrchardCore/ShellSettingsReleaseFilter.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.AspNetCore.Mvc.Filters;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace OrchardCore;
+
+public class ShellSettingsReleaseFilter : IResultFilter
+{
+    private readonly IShellContextReleaseService _shellContextReleaseService;
+
+    public ShellSettingsReleaseFilter(IShellContextReleaseService shellContextReleaseService)
+    {
+        _shellContextReleaseService = shellContextReleaseService;
+    }
+
+    public void OnResultExecuted(ResultExecutedContext context)
+    {
+        ShellScope.AddDeferredTask(async scope =>
+        {
+            // var shellContextReleaseService = scope.ServiceProvider.GetService<IShellContextReleaseService>();
+
+            await _shellContextReleaseService.ProcessAsync();
+        });
```

Comment:
I don't think complicating things here with a filter (and thus having an impact on all requests) makes sense. You can just add this to `ShellContextReleaseService` and it'll work. If there are multiple shell scopes started in a single request, which should be extremely rare, very deliberate, and not something that happens in the settings editor, then that still wouldn't warrant this, since you need the updated state of the shell (i.e. the restarted shell) in the next scope too (not just in a scope in the next request).

Furthermore, this doesn't need to be done from outside the service, since it can just keep this internal.

Reply:
I don't think I understand here. Yes I would love to see a better way this can be done without a filter. But this call has to be done per-request. Adding tried calling `ShellScope.AddDeferredTask(scope => _shellContextReleaseService.ProcessAsync());` during the app startup, but that does not call that task on every request.

Reply:
@Piedone I actually removed the filter altogether and added a default task that us executed before shell-disposing.